### PR TITLE
Support for numeric constraints

### PR DESCRIPTION
**Pull request by tachyons**

### CODE REVIEW

File: lib/jake/integer.ex [hunk 1]
```diff
@@ -1,7 +1,26 @@
...
+      end
+
+    max_bound =
+      case {max, exclusive_max} do
+        {nil, false} -> 9_007_199_254_740_991
+        {max, true} -> max - 1
+        {max, false} -> max
+      end
+
+    StreamData.integer(trunc(min_bound / multipleOf)..trunc(max_bound / multipleOf))
```
Comment (ananthakumaran) [inline lib/jake/integer.ex:23]: is trunc correct in all cases? could you think of any case where it might fail?

File: lib/jake/integer.ex [hunk 2]
```diff
@@ -1,7 +1,26 @@
...
+    min = Map.get(spec, "minimum")
+    max = Map.get(spec, "maximum")
+    exclusive_min = Map.get(spec, "exclusiveMinimum", false)
+    exclusive_max = Map.get(spec, "exclusiveMaximum", false)
+    multipleOf = Map.get(spec, "multipleOf", 1)
+
+    min_bound =
+      case {min, exclusive_min} do
+        {nil, false} -> -9_007_199_254_740_991
+        {min, true} -> min + 1
```
Comment (ananthakumaran) [inline lib/jake/integer.ex:12]: what if min is nil? when you pattern match, you have to consider all possibilities.

File: lib/jake/number.ex [hunk 3]
```diff
@@ -2,6 +2,47 @@ defmodule Jake.Number do
   def gen(spec) do
     min = Map.get(spec, "minimum", -9_007_199_254_740_991)
     max = Map.get(spec, "maximum", 9_007_199_254_740_991)
-    StreamData.integer(min..max)
+    exclusive_min = Map.get(spec, "exclusiveMinimum", false)
```
Comment (ananthakumaran) [inline lib/jake/number.ex:5]: this seems rather complicated and not sure if it's necessary.

for starter, I would move the generator to StreamUtil,

```elixir
def number(min, max) do
  StreamData.one_of([
   StreamData.float(..),
   StreamData.integer(..)
  ])
end
```

would try to de-duplicate the code between number & integer. I think the general logic is same, just that some function/constant would be different. For exclusive min max, we could just write a filter that filters out both values like

```elixir
def filter(list) do
  StreamData.filter(&Enum.member?(list, &1))
end
```

**Review by ananthakumaran — CHANGES_REQUESTED** [review]

File: lib/jake/integer.ex (hunk 2 above)
Comment (tachyons) [reply lib/jake/integer.ex:12]: if min is nil and exclusive_min is present, then that schema is invalid as per spec

File: lib/jake/number.ex (hunk 3 above)
Comment (tachyons) [reply lib/jake/number.ex:5]: But how do we differentiate between integer and float stream here

```
Jake.StreamUtil.number(1,22) |> Enum.take(10)
[1.0, 1.0, 19.375, 22.0, 12.8125, 22.0, 22.0, 15, 8.21875, 8]
```

Comment (ananthakumaran) [reply lib/jake/number.ex:5]: why do we need to differentiate? Both integer and float are acceptable as a number?

> The number type is used for any numeric type, either integers or floating point numbers.

Comment (tachyons) [reply lib/jake/number.ex:5]: Sorry, I thought you meant merging those 2

**Pull request closed without merging**